src/main/java/com/limechain/babe/Authorship.java

```diff
@@ -10,54 +15,194 @@
 import io.emeraldpay.polkaj.schnorrkel.VrfOutputAndProof;
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
+import lombok.extern.java.Log;
+import org.apache.commons.collections4.map.LinkedMap;
+import org.bouncycastle.jcajce.provider.digest.Blake2b;
 import org.javatuples.Pair;
 import org.jetbrains.annotations.NotNull;
 
 import java.math.BigInteger;
 import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
 
+@Log
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class Authorship {
 
-    public static BabePreDigest claimPrimarySlot(final byte[] randomness,
-                                                 final BigInteger slotNumber,
-                                                 final BigInteger epochNumber,
-                                                 final Schnorrkel.KeyPair keyPair,
-                                                 final long authorityIndex,
-                                                 final BigInteger threshold) {
+    public static BabePreDigest claimSlot(EpochState epochState, KeyStore keyStore) {
 
-        var transcript = makeTranscript(randomness, slotNumber, epochNumber);
+        var randomness = epochState.getCurrentEpochData().getRandomness();
+        var slotNumber = epochState.getCurrentSlotNumber();
+        var epochIndex = epochState.getEpochIndex();
+        var c = epochState.getCurrentEpochDescriptor().getConstant();
+        var authorities = epochState.getCurrentEpochData().getAuthorities();
 
-        Schnorrkel schnorrkel = Schnorrkel.getInstance();
-        VrfOutputAndProof vrfOutputAndProof = schnorrkel.vrfSign(keyPair, transcript);
-        byte[] vrfBytes = schnorrkel.makeBytes(keyPair, transcript, vrfOutputAndProof);
+        var indexKeyPairMap = getOwnedKeyPairsFromAuthoritySet(authorities, keyStore);
 
-        if (vrfBytes.length != 16) {
-            throw new IllegalArgumentException("VRF byte array must be exactly 16 bytes long");
+        BabePreDigest primarySlot = claimPrimarySlot(
+                randomness,
+                slotNumber,
+                epochIndex,
+                authorities,
+                c,
+                indexKeyPairMap
+        );
+
+        if (primarySlot != null) return primarySlot;
+
+        boolean authorSecondaryVrfSlot =
```

Comment:
Something is weird with the empty spaces here.

Reply:
When comparing two values, I always try to keep them on one line to make the comparison more intuitive. But I see that I didn't manage to do that properly here.